### DISCUSSION TITLE
Virtual Machine permission issues

### DISCUSSION THREAD

**Discussion by GioeleLiva** [bitcraze/discussions#1535] [Q&A] [answered]
Hello community, 
i'm following the getting started guide for academic purposes, i downloaded and run the 2023 version of bitcrazeVM on virtualbox (on a linux machine); 
the dongle connects, and the clients works as intended, but i'm getting issues while doing little demos in python scrypts: the exact error is 
"Got link error callback [Too many packets lost] in state [1]
Traceback (most recent call last):
  File "/home/bitcraze/myprojects/connect_log_param.py", line 21, in <module>
    with SyncCrazyflie(uri, cf=Crazyflie(rw_cache='./cache')) as scf:
  File "/home/bitcraze/.local/lib/python3.10/site-packages/cflib/crazyflie/syncCrazyflie.py", line 119, in __enter__
    self.open_link()
  File "/home/bitcraze/.local/lib/python3.10/site-packages/cflib/crazyflie/syncCrazyflie.py", line 95, in open_link
    raise Exception(self._error_message)
Exception: Too many packets lost"
but seems like the system is having trouble creating the json file in /cache;
i tried doing this with and without sudo, the only difference is with sudo i get cflib: library not found;
it feels like the scrypt has no permission to write, but i chmodded everything;
any idea what the cause of the problem is?
thanks in advance

**Comment by gemenerik** [comment, marked as answer]
It appears that you are following the "Step-by-Step: Connecting, logging and parameters" user guide, correct?
Please verify that the uri in the code matches that of your CF.

```python
uri = 'radio://0/80/2M/E7E7E7E7E7
```

If that's not it:
Since you got the client working, can you check if you are trying to use the same cflib backend as the client does?

As a workaround I would recommend using a native install.

**Reply by GioeleLiva** [reply]
Ok i had to change to /250K/E7E7E7E7E7, since in a previous test (on another machine) i lowered the frequence  to 250K to handle major distances, i didn't know that was a problem, thanks a lot!
